staging/src/k8s.io/cloud-provider/plugins.go

```diff
@@ -40,7 +40,7 @@ var (
 		external bool
 		detail   string
 	}{
-		{"gce", false, "The GCE provider is deprecated and will be removed in a future release. Please use https://github.com/kubernetes/cloud-provider-gcp"},
+		{"gce", true, "The GCE built-in cloud provider was removed. Please use https://github.com/kubernetes/cloud-provider-gcp"},
```

Comment:
given that `gce` is the only value in this structure, it makes me wonder if the other formerly in-tree providers should be here as well? or should it just be empty?

perhaps we should change the logic to print a warning for any cloud provider that is specified?

Reply:
The structure of the deprecatedCloudProviders was changed in https://github.com/kubernetes/kubernetes/pull/63686. It introduces two new fields: `external` and `detail`. 

- `external` is a boolean field that indicates whether the cloud provider is moved to an external repository and the in-tree cloud provider becomes deprecated.
- `detail` is a string field that provides more information about the deprecation.

The items in the `deprecatedCloudProviders` should be removed simultaneously when the corresponding cloud provider is removed from the in-tree. However, when they started the deprecation process, some cloud providers did not set the `external` field to `true`.

So I may need to remove the item from the `deprecatedCloudProviders`. Should I cherry-pick it to the 1.31 branch?

@dims @elmiko

Reply:
i don't think it's a huge deal, but if we can backport without too much headache i think we should.

Reply:
Removed. I keep the empty list because I'm unsure whether it is right to clean up the code in 1.32. 

- the DisableCloudProviders feature-gate was GA in 1.31
- KEP https://github.com/kubernetes/enhancements/tree/master/keps/sig-cloud-provider/2395-removing-in-tree-cloud-provider:  Removal of all in-tree code no earlier than one release after GA release.